src/misc_utils.jl

```diff
@@ -197,3 +197,30 @@ function get_differential_vars(f, u)
         end
     end
 end
+
+"""
+Helper to evaluate
+    f(innertmp + γ̂⋅z, p, t + c⋅dt)
+in the nonlinear solver when z is an array partition and γ̂ is the scaling parameter.
+Note that γ̂ is different from the γ in the Newmark-β method!
+"""
+struct ArrayPartitionNLSolveHelper{T}
+    γ₁::T
+    γ₂::T
+end
+
+function Base.:*(γ::ArrayPartitionNLSolveHelper{T1}, z::ArrayPartition{T2, <: Tuple{<:Any, <:Any}}) where {T1, T2}
+    ArrayPartition(γ.γ₁*z.x[1], γ.γ₂*z.x[2])
+end
+
+function Base.:*(γ::ArrayPartitionNLSolveHelper{T1}, z::Vector{T2}) where {T1, T2}
+    ArrayPartition(γ.γ₁*z, γ.γ₂*z)
+end
+
+Base.:*(γ::ArrayPartitionNLSolveHelper{T}, scalar::T) where T = scalar*γ
+
+function Base.:*(scalar::T, γ::ArrayPartitionNLSolveHelper{T}) where T
+    ArrayPartitionNLSolveHelper(scalar*γ.γ₁, scalar*γ.γ₂)
+end
+
+Base.inv(γ::ArrayPartitionNLSolveHelper) = 1/(γ.γ₁+γ.γ₂)
```

Comment:
Because I am not sure how to integrate a pair of weights with the W transformation. This is basically a hotfix to make the example work at all.

If you take another look at the perform step function you will see what I am trying to achive. Basically the nonlinear solver interface assumes that $\gamma$ is a scalar, but for Newmark methods we have one gamme for the first part of the solution (block 1 in the associated ArrayPartition) and one gamma for block 2).

Reply:
I abandoned this idea for now, as it needed too many hacks.